proposed/http-message.md

```diff
@@ -332,9 +505,11 @@ interface IncomingRequestInterface extends MessageInterface
      * purposes of how duplicate query parameters are handled, and how nested
      * sets are handled.
      *
-     * @return array
+     * @param array $query Array of query string arguments, typically from
+     *     $_GET.
+     * @return void
      */
-    public function getQueryParams();
+    public function setQueryParams(array $query);
```

Comment:
Correct -- this all part of that "Logical/Usage" hybrid @Crell posted on the ML about; it's only in the `ServerRequest`. I agree -- if this had been part of the regular request interface, I'd want 2-way binding!

One last point: an implementation can always choose immutability. That's one reason I think mutability is important here -- it allows the most common scenarios, and those who want immutability can always have the setters throw exceptions in their implementations or decorators. :)